[TODO] Avoid 290 fix releases

DanMesh opened the pull request:
… tag

* Do not assume the release name is exactly of the format 'v#.#.#'.
* All tags will create releases.

#### Add a link of the Jira Task
Jira Task: *URL*

#### Does this PR need to be backported to the stable release?
<!-- Uncomment one of the following two lines -->
<!-- YES! My PR won't be merged until I made also the PR to the stable branch -->
<!-- NO! No other PR is needed -->

#### Does this PR need to update any documentation (readme, confluence, gitbook, etc.)?
<!-- Uncomment one of the following two lines -->
<!-- YES! My PR won't be merged until I provide a link to the updated docs -->
<!-- NO! -->

#### Describe your solution
*A clear and concise description of what you have implemented.*

#### Test data / coverage
How was this tested? What Platforms were covered?
<!-- * Skynode -->
<!-- * Imx8mm dev kit -->
<!-- * RB5 -->
<!-- * Laptop -->

#### Screenshot/cast


#### Additional information

TSC21 reviewed (changes requested):
This is an anti-pattern for releases. Let's please bump the patch version instead of doing this.

DanMesh commented:
@TSC21 This doesn't change the release workflow in any way. All it's doing is ensuring that you don't have releases unexpectedly replacing each other (as I experienced yesterday).

Andreea-L commented:
Sorry for the review delay, I kinda lost this PR in my list. :sweat_smile: 
I see both sides of the argument, since on the one hand every developer _should_ strive to use the given major.minor.patch versions as much as possible, without the need for extra. -betaX suffixes are very annoying to have to scroll through in a release list (see AOS) and if they can be explicitly avoided/forbidden, then all the better.

On the other hand, if you know you'll have very manufacturer-specific patches that cannot go into any other release but `v1.2.3-manufacturer`, then this overwriting of artifacts is a clear limitation. 

Overall though, I agree with the main point of the PR. Most people would expect the CI to build a release with the exact name of the tag they just created. But then the tags need to be more strictly groomed. I think the requirement for the 'v' should remain in place though. Otherwise if people start making tags without it, all the previous 'v' tags will get sorted underneath and that's also confusing.

Just my two cents :) If there are any concerns specific to this project that I'm not familiar with, I'm happy to stand corrected ofc

DanMesh commented:
Thanks @Andreea-L, those are some very good points. I agree that we should be aiming for a well-defined practice for how releases are named. For something like the Autopilot Manager, I don't think it's worth trying to automate this, and would rather "encourage" developers to stick to the rules :)

For now, I would say it's enough to at least prevent the release overwriting. You make a good point regarding the value of the leading `v` so I've added that requirement back, and rebased due to merge conflicts.
